### DISCUSSION TITLE
Keine Verbindung zur Luxtronik

### DISCUSSION THREAD

**Discussion by jensrossbach** [cbrosius/luxtronik_v1#26] [Q&A] [answered]
Hallo @CBrosius,

ich versuche schon seit geraumer Zeit, meine Wärmepumpe in Home Assistant einzubinden. Dazu habe ich ein ähnliches Setup wie von dir in diesem Projekt hier beschrieben, aber ich bekomme es einfach nicht hin.

Der einzige Unterschied zu deinem Setup ist, dass ich das MAX3232-Modul direkt an die RS232-Schnittstelle der Luxtronik angeschlossen habe und dann mit einem Telefonkabel (4 Adern - VCC, GND, RX, TX) nach außen gehe und dort auf den ESP32.

Problem: Es kommen keine Antworten auf die Befehle an die Wärmepumpe (z.B: "1100\r\n") zurück.

Ich habe schon zig-mal die seriellen Parameter (Baudrate usw.) überprüft und die stimmen definitiv. Die Schnittstelle der Luxtronik ist auch korrekt konfiguriert (Diagnose - Standard). Das konnte ich auch überprüfen, indem ich einen PC über eine serielle Verbindung an die Luxtronik anschlossen habe und mit PuTTY die Befehle schickte - dann bekam ich die erwarteten Antworten zurück.

Ich habe auch eine Simulation der Luxtronik mittels eines kleinen Python-Skripts geschrieben. Dieses Skript schickt einfach Beispielwerte auf die verschiedenen Befehle hin zurück. Wenn ich meine ESP32-basierte Hardware mit dem PC, auf dem die Simulation läuft, verbinde (also das MAX3232-Modul an den PC statt an die Wärmepumpe anschließe), funktioniert es auch (die Befehle, die der ESP32 schickt, kommen in der Simulation an und die Antwortdaten kommen im ESP32 an). Nur wenn ich die Hardware mit der Wärmepumpe verbinde, funktioniert es nicht mehr. Der einzige Unterschied zwischen funktionierenden und nicht funktionierenden Aufbauten ist ein RS232-USB-Konverter, den ich benötige, da mein PC keine RS232-Schnittstelle mehr hat. Ansonsten ist alles identisch.

Hast du vielleicht noch eine Idee, was ich falsch machen könnte? Ich komme ich hier einfach nicht mehr weiter.

Danke und Gruß, Jens

**Comment by optimismus** [comment, marked as answer]
Hallo Jens, auch wenn ich nicht der Adressierte bin: Ich tippe auf das Kabel/die physische, serielle Verbindung. Ich hatte das auch, weil manche Seriellgeräte automatisch Senden/Empfangen tauschen. Hast du schon mal Tx und Rx getauscht?
Axel

**Reply by jensrossbach** [reply]
ja, hab ich auch schon versucht, leider auch ohne Erfolg. Ich habe aber nur auf der TTL-Seite getauscht, also zwischen ESP32 und MAX3232, da ich ja auf der anderen Seite den Stecker des MAX3232-Moduls direkt an die RS232-Schnittstelle der Luxtronik angeschlossen habe (ohne Zwischenkabel). Spielt es eine Rolle auf welcher Seite man vertauscht?

**Reply by jensrossbach** [reply]
Jetzt funktioniert es! Ich habe mir einen Nullmodem-Adapter besorgt und zwischengeschaltet, das war die Lösung. Danke nochmal für den Hinweis, @optimismus!

**Reply by Arni1000** [reply]
Hallo,
hat sich zwischenzeitlich die Yaml Datei für neuere ESPhome Versionen geändert? Wenn ich den ESP32 mit der gespeicherten Datei mit ESPhome flashen möchte wird nur eine Fehlermeldung ausgegeben. 
Die Meiner Meinung nach sagt dass custom component nicht mehr unterstüzt wird.

INFO ESPHome 2025.2.0
INFO Reading configuration /config/esphome/luxtronik-v1-esp32.yaml...
Failed config

sensor.custom: [source /config/esphome/luxtronik-v1-esp32.yaml:61]
  
  The "custom" component has been removed. Consider conversion to an external component.
  https://esphome.io/guides/contributing#a-note-about-custom-components.
  platform: custom
  lambda: |-
    auto luxtronik_v1 = new luxtronik_v1_sensor(id(uart_bus));
    App.register_component(luxtronik_v1);
    return {
      luxtronik_v1->temp_VL,         
      luxtronik_v1->temp_RL,         
      luxtronik_v1->temp_RL_Soll,
      luxtronik_v1->temp_Heissgas,   
      luxtronik_v1->temp_Aussen,     
      luxtronik_v1->temp_BW,

**Comment by cbrosius** [comment]
Ich habe gesehen, das sich mit ESPHome 2025.2 da was geändert hat. Werde
aber frühestens nächste Woche dazu kommen mir das anzusehen. Wenn möglich
aktuell eine der älteren Varianten benutzen.

Arni1000 ***@***.***> schrieb am Sa., 22. Feb. 2025, 12:39:

**Reply by Arni1000** [reply]
Danke Dir, vielleicht kannst du mir die yaml zukommen lassen und eventuell die sensor.h. 
Gruß

**Reply by cbrosius** [reply]
Es gibt jetzt eine neue Version. Schau Dir die mal an. Ich denke mit der aktualisierten YAML ist es jetzt deutlich einfacher.
Gruß,
Christian...

**Reply by Arni1000** [reply]
Vielen Dank für die Info. Ich bin zurzeit ans Bett gefesselt und kann es momentan nicht ausprobieren. Trotzdem hätte ich noch eine Frage zur Firmware. Bei meiner WP ist die V2.20 installiert und ich meine mal in einem anderen Projekt (FHEM?) gelesen zu haben, dass da eine neuere FW installiert sein müsste.
Gruß

**Reply by cbrosius** [reply]
Hmm, gute Frage. Ich habe noch keine Erfahrung mit anderen Versionen als v2.33. Aber einen Versuch wird es schon wert sein.

**Reply by approx3145** [reply]
habe auch die V2.20

**Reply by Arni1000** [reply]
Hallo, das Programmieren des ESP32 hat jetzt schonmal funktioniert. Alle Sensoren sind als Entitäten verfügbar aber ich bekomme keine Rückantwort von der Luxtronik, Ich habe eigentlich alles durchprobiert. TX/RX getauscht, Nullmodemadapter zwischengeschaltet so wie es jensrossbach gemacht hat, verschiedene Bautraten eingestellt, und das ganze noch mit einem ESP8266 D1 mini probiert leider alles ohne Erfolg. Diagnose-Schnittstelle ist auf Standard eingestellt.  Ich lege das Projekt jetzt erstmal auf Eis und besorge mir nochmal einen anderen ESP32. Vielleicht ist es ja ein Hardware Problem.

@approx3145 hast du mit der V2.20 eine Verbindung herstellen können bzw. bekommst du Rückmeldungen von der Luxtronik?
